pallets/msa/src/lib.rs

```diff
@@ -1357,6 +1360,67 @@ impl<T: Config> Pallet<T> {
 		Ok(result)
 	}
 
+	/// Converts an MSA ID into a synthetic Ethereum address (raw 20-byte format) by
+	/// taking the last 20 bytes of the keccak256 hash of the following:
+	/// [0..1]: 0xD9 (first byte of the keccak256 hash of the domain prefix "Frequency")
+	/// [1..9]:   u64 (big endian)
+	/// [9..41]:  keccack256("MSA Generated")
+	pub fn msa_id_to_eth_address(id: MessageSourceId) -> H160 {
+		/// First byte of the keccak256 hash of the domain prefix "Frequency"
+		/// This "domain separator" ensures that the generated address will not collide with Ethereum addresses
+		/// generated by the standard 'CREATE2' opcode.
+		const DOMAIN_PREFIX: u8 = 0xD9;
+
+		lazy_static! {
```

Comment:
I'm unsure about the behavior of `lazy_static` inside a scope such as a function.  I've only seen it in global scope.

Reply:
Pretty sure it just limits the scope of the the resulting `static ref` variable to the enclosing function. I prefer to keep things scoped as minimally as possible.